numpy/__init__.pxd

```diff
@@ -942,6 +946,91 @@ cdef extern from *:
     /* NumPy API declarations from "numpy/__init__.pxd" */
     """
 
+cdef extern from *:
+    ctypedef class numpy.floating [object PyObject]:
+        pass
+
+
+cdef inline bint is_integer_object(object obj):
+    """
+    Cython equivalent of
+
+    `isinstance(val, (int, np.integer)) and not isinstance(val, bool)`
+
+    Cython cannot optimize this into a C call on its own because
+    np.integer is not available as a C type.
+
+    Parameters
+    ----------
+    val : object
+
+    Returns
+    -------
+    is_integer : bool
+
+    Notes
+    -----
+    This does not counts np.timedelta64 objects as integers.
+    """
+    return (not PyBool_Check(obj) and PyArray_IsIntegerScalar(obj)
```

Comment:
Note the definition of this function is just:

https://github.com/numpy/numpy/blob/6ee49178517088966e63c2aedf6a8a5779ad5384/numpy/core/include/numpy/ndarrayobject.h#L51-L52

Reply:
yah, its excluding bool and td64 that makes this useful

Reply:
Sure, but after that change it no longer uses any numpy C API functions, so it probably doesn't belong in `numpy/__init__.pxd` any more.